src/sentry/monitors/system_incidents.py

```diff
@@ -166,6 +177,258 @@ def get_clock_tick_volume_metric(tick: datetime) -> float | None:
+    """
+    Given a clock tick timestamp determine based on the historic tick volume
+    metrics, and historic tick anomaly decisions, a DecisionResult.
+
+    Once there is an INCIDENT future ticks will continue to be marked as
+    INCIDENT until we begin detecting metrics below the abnormal metric
+    threshold, at which point the system will recover after
+    MONITOR_TICK_DECISION_WINDOW ticks below the abnormal threshold.
+
+    This function will update previous decisions for earlier ticks detected as
+    ABNORMAL to either NORMAL or INCIDENT.
+    """
+    # Alias TickAnomalyDecision to improve code readability
+    Decision = TickAnomalyDecision
+
+    if not options.get("crons.tick_volume_anomaly_detection"):
+        return DecisionResult(Decision.NORMAL)
+
+    redis_client = redis.redis_clusters.get(settings.SENTRY_MONITORS_REDIS_CLUSTER)
+
+    # The clock has just ticked to the next minute. Look at the previous tick
+    # and decision metrics.
+    past_ts = tick - timedelta(minutes=1)
+
+    past_window_ts_keys = [
+        _make_reference_ts(past_ts - timedelta(minutes=delta))
+        for delta in range(0, MONITOR_TICK_DECISION_WINDOW)
+    ]
+
+    # Fetch histories for metrics and the last decision together. Window
+    # timestamps are reversed so the oldest metric is last.
+    redis_keys = chain(
+        (MONITOR_TICK_METRIC.format(ts=ts) for ts in reversed(past_window_ts_keys)),
+        (MONITOR_TICK_DECISION.format(ts=ts) for ts in [past_window_ts_keys[0]]),
+    )
+
+    histories = redis_client.mget(redis_keys)
+
+    # Tick metrics are the first MONITOR_TICK_DECISION_WINDOW values
+    tick_metrics = [float(value) if value else None for value in histories[:-1]]
+
+    # The last decision is the last value fetched
+    if histories[-1] is not None:
+        last_decision = Decision.from_str(histories[-1])
+    else:
+        # By default the previous decision is used. If there was no previous
+        # decision we can only assume things are operating normally
+        last_decision = Decision.NORMAL
+
+    # Pre-compute decisions for the history of metrics
+    metric_decisions = [_metric_decision(metric) for metric in tick_metrics]
+
+    def make_decision(
+        decision: TickAnomalyDecision,
+        transition: AnomalyTransition | None = None,
+    ) -> DecisionResult:
+        decision_key = MONITOR_TICK_DECISION.format(ts=_make_reference_ts(tick))
+        redis_client.set(decision_key, decision)
+        redis_client.expire(decision_key, MONITOR_VOLUME_RETENTION)
+
+        return DecisionResult(decision, transition)
+
+    def metric_matches(decision: TickAnomalyDecision) -> Generator[bool]:
+        return (d == decision for d in metric_decisions)
+
+    # ENTERING an abnormality
+    #
+    # If we've detected an anomaly and we're not already in an incident or
+    # anomalous state, mark this tick as anomalous.
+    if last_decision == Decision.NORMAL and metric_decisions[-1] == Decision.ABNORMAL:
```

Comment:
We use the last `metric_decision` a lot here. Is there a more specific name we could use?

Reply:
I've updated this. I'm now calling it `last_metric`